tests/Monolog/HandlerTest.php

```diff
@@ -49,11 +60,16 @@ public function handleDataProvider(): iterable
                 'context' => [],
                 'extra' => [],
             ],
-            [
-                'level' => Severity::debug(),
-                'message' => 'foo bar',
-                'logger' => 'monolog.channel.foo',
-            ],
+            (static function (): Event {
+                $event = Event::createEvent();
```

Comment:
If you move the instantiation of the event before the `yield` statement we can avoid creating all these self-invokable lambdas which makes the code less readable